### PR TITLE
[WIP] Indentation rules should be made clearer in list comprehsions

### LINKED ISSUE #4234: Indentation rules should be made clearer in list comprehsions
For elmish newcomers it's sometimes hard to understand the indentation rules. And compiler error messages are not helpful:

    let div _ _ = 1

    let subView _ _ = [1; 2]

    let view1 model dispatch =
       [    
            div [] [
               subView model dispatch
            ]
       ]

    let view2 model dispatch =
       [    
            div [] 
            [
               subView model dispatch
            ]
       ]

    let view3 model dispatch =
       [   
            div [] 
                [
                   subView model dispatch
                ]
       ]

view1 and view3 are ok. view2 gives:

image

### PR DESCRIPTION
**Pull request by forki**
fixes #4234

### CODE REVIEW

**Comment by forki** [comment]
image

@isaacabraham would that make sense?

**Comment by enricosada** [comment]
`s/second expression/other expression in list`

**Comment by isaacabraham** [comment]
Hmmm, I'd consider changing some of the wording e.g. "This may be resolved by indenting the second expression further", but taking a step back, what's the cause of the error:

```fsharp
[
    div [] // resolves to a -> int?
    [
        subview model dispatch
    ] // resolves to b list?
]
```
Is this caused because the second line is not indented in line with the first line's `[]` i.e. it causes the first line to be treated as a partially applied function of 'a -> int` (why int?) and then the second bit is a standalone list? If so... I'm not sure how best to explain this unless you could somehow know for sure that the first expression is a function that expects an argument which is of the same type as the second expression?

**Comment by forki** [comment]
yes we checked that first thing is indeed a function

**Comment by isaacabraham** [comment]
Cool. If so - then perhaps you could go even further:

```
All elements of a list constructor expression must have the same type. This expression was expected to have type '%s', but here has type '%s'. This could be caused because you have not indented the second expression sufficiently, causing the previous expression to be evaluated as a partially applied function. Indenting the second expression further may resolve this.
```

By this I'm trying to get something that explains *why* the error occurs, and not just how to fix it. But, might be too wordy.

**Comment by cartermp** [comment]
@dotnet-bot  test this please

**Comment by cartermp** [comment]
@dotnet-bot test this please

**Comment by cartermp** [comment]
@forki It would be good to re-evaluate indentation for Elmish now that more "undentation" is supported in [F# 4.5](https://github.com/fsharp/fslang-design/blob/master/FSharp-4.5/FS-1054-undent-list-args.md)

I don't think that one makes this particular issue better, but it would be good to at least confirm that it didn't regress what already exists somehow

**Comment by dsyme** [comment]
Closing old PR
